### PR TITLE
Fix Texture Cache slow down and crashes

### PR DESCRIPTION
**Pull request by orbea**
I experienced an issue on Slackware64-current, nouveau, GLideN64 and the game Beetle Adventure Racing where after playing for a while the game would begin to lag progressively worse until it would clear the texture cache and potentially bring my entire system down.

The problem was that Beetle Adventure Racing is capable of producing infinite textures and that GLideN64 had a max cache limit of 128,000 textures. This is problematic because some video drivers such as nouveau convert every single one of these textures into a 128 KB "large" page which would require 16 GB of vram where I have 3 GB.  After discussing this with the nouveau developers at #nouveau @ freenode it was suggested to have this limit as 16,384 instead which would require 2 GB of vram and is much saner. 

This avoided the slow down and crashes, but introduced an annoying issue where the game would freeze (Hiccup) momentarily every time it hit that limit and cleared the cache. To combat this loganmc10 suggested to clear only the oldest textures instead of the entire cache. This works wonderfully and solves all issues encountered.

So in short, this will fix slow downs, crashes and hiccups with GLideN64 and Beetle Adventure Racing while using some linux video drivers.

Please read this issue report that this PR will fix for more information.
https://github.com/loganmc10/GLupeN64/issues/56

### CODE REVIEW

**Comment by gonetz** [comment]
Thanks for well explained commit. I tried do something like this when worked on #795, but the game generates large number of new textures each frame, so removing them by one was too slow. It was better to clear cache by the price of performance hiccup. Good if this fix works well for BAR.

**Comment by orbea** [comment]
I've tested with NBA Hangtime and seems to require a more complete solution for that game. With this change its slow, but if the entire cache is cleared instead it suffers from being both slow and hiccuping. At least BAR seems to work great now.

**Comment by orbea** [comment]
I guess NBA Hangtime is much faster if the limit is even lower, but even if the limit is reduced to something like 1,000 it still often dips to 40 fps.

**Comment by loganmc10** [comment]
You could always delete them in a batch. Like when it hits the 16384 limit, delete the 100 oldest textures, instead of wiping the whole cache or doing it one at a time.

**Comment by orbea** [comment]
I think its more that the game creates textures far faster than nouveau can handle. I'm not sure about other video drivers, mupen64plus-libretro with glide64 is pretty slow too.

**Comment by gonetz** [comment]
> You could always delete them in a batch. Like when it hits the 16384 limit, delete the 100 oldest textures, instead of wiping the whole cache or doing it one at a time.

Sounds like a good optimization trick.
